src/Servers/Kestrel/Core/src/Internal/Http3/Http3PeerSettings.cs

```diff
@@ -1,12 +1,35 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3
 {
     internal class Http3PeerSettings
     {
-        internal const uint DefaultMaxFrameSize = 16 * 1024;
+        // Note these are protocol defaults, not Kestrel defaults.
+        public const uint DefaultHeaderTableSize = 0;
+        public const uint DefaultMaxRequestHeaderFieldSize = uint.MaxValue;
+
+        public uint HeaderTableSize { get; internal set; }
+        public uint MaxRequestHeaderFieldSize { get; internal set; }
+
+        // Gets the settings that are different from the protocol defaults (as opposed to the server defaults).
+        internal IList<Http3PeerSetting> GetNonProtocolDefaults()
+        {
+            var list = new List<Http3PeerSetting>(1);
```

Comment:
Why 1? Size looks dynamic so IMO don't set an initial capacity

Reply:
The same was done in [HTTP/2](https://github.com/dotnet/aspnetcore/blob/3b7cdc166aa3c9733fa60b56d91fc0fff9b11652/src/Shared/ServerInfrastructure/Http2/Http2PeerSettings.cs). I believe it's because the default number of settings to send is one for HTTP/2. The same is for HTTP/3 here; the only setting that isn't default is the MaxRequestHeaderFieldSize.

Reply:
I could set the capacity to 2 to make sure it will always fit?

Reply:
Add a comment explaining why 1 is used. Is this copied from HTTP/2 settings? If so then add the comment there as well.